File: code/java/minPathSum.java
class Solution {
    public int minPathSum(int[][] grid) {
        int[][] minPath = new int[grid.length][grid[0].length];
        minPath[0][0] = grid[0][0];
        for(int i=0;i<grid.length;i++)
            for(int j=0;j<grid[0].length;j++){
                int a,b;
                if(i==0 && j==0)
                    continue;
                if(i>0)
                    a = minPath[i-1][j];
                else
                    a = Integer.MAX_VALUE;
                if(j>0)
                    b = minPath[i][j-1];
                else
                    b = Integer.MAX_VALUE;
                minPath[i][j] = Math.min(a,b) + grid[i][j];
            }
        return minPath[grid.length-1][grid[0].length-1];
    }
    
}

File: code/python/minPathSum.py
class Solution(object):
    def minPathSum(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        minPath = []
        temp = []
        for i in range(len(grid[0])):
            temp.append(float('inf'))
        for i in range(len(grid)):
            minPath.append(temp)
        minPath[0][0] = grid[0][0]
        for i in range(len(minPath)):
            for j in range(len(minPath[0])):
                if i + j == 0:
                    continue
                if i > 0:
                    a = minPath[i-1][j]
                else:
                    a = float('inf')
                if j > 0:
                    b = minPath[i][j-1]
                else:
                    b = float('inf')
                minPath[i][j] = min(a,b) + grid[i][j]
        return minPath[-1][-1]
